source/groupBy.js

```diff
@@ -44,7 +44,10 @@ import reduceBy from './reduceBy.js';
  *      // }
  */
 var groupBy = _curry2(_checkForMethod('groupBy', reduceBy(function(acc, item) {
+  if (acc == null) {
+    acc = [];
+  }
   acc.push(item);
   return acc;
-}, [])));
+}, null)));
```

Comment:
Mmmh I think the change introduced in https://github.com/ramda/ramda/commit/9b5d8925f23f0f44a2e656bfd0cf23ff30031794 were possible breaking changes:

```none
[].reduce(() => 42, null);
//=> null
[].reduce(() => 42, []);
//=> []
```

So reverting this definitely fixes the bug? Your repro case doesn't have empty lists as far as I can tell so I don't see an obvious connection here unless I'm looking at this from the wrong angle?

Reply:
The connection insofar as I saw it was that in [reduceBy](https://github.com/ramda/ramda/blob/master/source/reduceBy.js#L58)  - not so much a problem of passing an empty list, but the fact that the list accumulator is being re-cloned for each group in a way that's memoized and holds onto the original values.

Reverting the change makes the tests pass for `groupby`, but only for `groupBy`, though, so I suspect this may be ignoring other places where the same bug is showing up, which may need further digging.